Improvement: /shnavigate convenience

rueblimaster opened the pull request:
## What

- Added "shnav" as alias for "shnavigate".
- Made it navigate to searched spot instantly if only 1 result matched
  - Added a toggle in config
- Merged the `AreaNavigationConfig` and the `PathfindConfig` into a new `NavigationConfig`, where the new toggle was added


## Changelog Improvements
+ `/shnavigate` now instantly navigates if there is only one match. - rueblimaster
    * Can be disabled in config.
+ Added `/shnav` as alias to `/shnavigate`.  - rueblimaster

rueblimaster committed:
add "shnav" as shnavigate alias

rueblimaster committed:
remodel navigation config stuff

it's now based with the pathfinding stuff in 1 category

rueblimaster committed:
add config fixes

rueblimaster committed:
small code improvements

rueblimaster committed:
implement auto start

rueblimaster committed:
added allowInstaNavigate config toggle

NopoTheGamer commented on src/main/java/at/hannibal2/skyhanni/features/misc/pathfind/NavigationFeedback.kt:
```diff
@@ -80,4 +83,9 @@ object NavigationFeedback {
+
+    @HandleEvent
+    fun onConfigFix(event: ConfigUpdaterMigrator.ConfigFixEvent) {
+        event.move(1235345, "features.misc.pathfinding", "features.misc.navigation.pathfinding")
```
?

rueblimaster replied:
ye mb about the number

NopoTheGamer commented on src/main/java/at/hannibal2/skyhanni/features/misc/IslandAreas.kt:
```diff
@@ -160,19 +162,19 @@ object IslandAreas {
-                }
-            }
+//                 }
+//             }
```
?

rueblimaster replied:
@NopoTheGamer 
the line that was commented out made the whole thing I commented out useless, so I also commented that part out.
idk what is the expected way to go in such situations but ye

hannibal002 replied:
> idk what is the expected way to go in such situations but ye

not having commented out code

NopoTheGamer commented on src/main/java/at/hannibal2/skyhanni/features/misc/IslandAreas.kt:
```diff
@@ -320,7 +322,13 @@ object IslandAreas {
+    @HandleEvent
+    fun onConfigFix(event: ConfigUpdaterMigrator.ConfigFixEvent) {
+        event.move(1235345, "features.misc.areaNavigation.areaPathfinder", "features.misc.navigation.areaPathfinder")
+        event.move(1235345, "features.misc.areaNavigation", "features.misc.navigation.islandAreas")
```
?

rueblimaster replied:
ye mb about the number

NopoTheGamer commented on src/main/java/at/hannibal2/skyhanni/config/features/misc/navigation/NavigationConfig.kt:
```diff
@@ -0,0 +1,28 @@
+    @Expose
+    @ConfigOption(name = "Allow Instant Navigate", desc = "Starts navigating instantly upon shnavigate with only one match.")
+    @ConfigEditorBoolean
+    val allowInstaNavigate: Boolean = true
```
really should be allowInstantNavigation

NopoTheGamer reviewed (changes requested):
havent tested in game

rueblimaster committed:
Merge branch 'beta' into shnavigate-improvements

rueblimaster committed:
use new config version

rueblimaster committed:
Rename allowInstaNavigate to allowInstantNavigation

rueblimaster committed:
fix

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/misc/IslandAreas.kt:
```diff
@@ -42,7 +43,8 @@ import kotlin.time.Duration.Companion.milliseconds
 @SkyHanniModule
 object IslandAreas {
-    private val config get() = SkyHanniMod.feature.misc.areaNavigation
+    private val pathfingerConfig get() = SkyHanniMod.feature.misc.navigation.areaPathfinder
```
pathfinger

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/config/features/misc/navigation/NavigationConfig.kt:
```diff
@@ -0,0 +1,28 @@
+
+class NavigationConfig {
+    @Expose
+    @ConfigOption(name = "Allow Instant Navigation", desc = "Starts navigating instantly upon shnavigate with only one match.")
```
mark the command in yellow, add / prefix.
Move to the bottom of the file

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/config/features/misc/navigation/NavigationConfig.kt:
```diff
@@ -0,0 +1,28 @@
+    @Expose
+    @ConfigOption(name = "Area Path Finder", desc = "Helps navigate to different areas on the current island.")
+    @Accordion
+    val areaPathfinder: AreaPathfinderConfig = AreaPathfinderConfig()
```
can we move island areas inside area path finder? i dont see why the settings should be split up

rueblimaster replied:
@hannibal002 the IslandAreasConfig stuff looked like it's designed to be standalone and not directly tied to a specific feature, but that's something I'd be open to change.
How do you want it?

rueblimaster replied:
@hannibal002 I just tried it and I find it kinda stupid, because you have 1 config that has 1 feature, with a enable toggle but then you have other features also inside of that Accordion. I don't think that's really viable.
It would look about like this: https://github.com/rueblimaster/SkyHanni/tree/shnavigate-merged-areas

hannibal002 replied:
> @hannibal002 the IslandAreasConfig stuff looked like it's designed to be standalone and not directly tied to a specific feature.

Its designed to the three features implemented in "island areas" class: "show in world", "title on enter" and "list of areas around you".

all options from AreaPathfinderConfig and IslandAreasConfig are only used inside IslandAreas. so logically they all connect. area pathfinder config could be a part inside island area config?
the option includeSmallAreas can belong into your new class navigation config, but everything else is one compact feature with three toggles. note how the other two settings in island area config (show in world and title on enter) are not used for /shnavigate or navigation features in general.

also i thin kthe island areas class needs to be split up in backend logic and those three features, similar to damage indicator.

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/misc/pathfind/NavigationHelper.kt:
```diff
@@ -132,6 +144,7 @@ object NavigationHelper {
     fun onCommandRegistration(event: CommandRegistrationEvent) {
         event.registerBrigadier("shnavigate") {
             description = "Using path finder to go to locations"
+            aliases = listOf("shnav")
```
how necessary is this? if i type /shn and press tab, the name navigate gets auto completed already.

rueblimaster replied:
@hannibal002 that is to be discussed, it really depends on what the mods policy wants to be
also this is because of this suggestion: https://discord.com/channels/997079228510117908/1447378507289198775
what is your opinion on the matter?

hannibal002 replied:
<img width="298" height="74" alt="image" src="https://github.com/user-attachments/assets/3e6326ba-18f3-4dc2-9e82-e2fc28ed3d2e" />

this was the discussion. but sure, one more command doesnt hurt ig.

rueblimaster replied:
okay, so I'm leaving it as is

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/misc/pathfind/NavigationHelper.kt:
```diff
@@ -66,32 +68,42 @@ object NavigationHelper {
         componentText.send(messageId)
     }
 
+    private fun sendNavigateMessage(name: String, goBack: () -> Unit) {
```
can we inline this function?

rueblimaster replied:
ye sure

hannibal002 commented on src/main/java/at/hannibal2/skyhanni/features/misc/pathfind/NavigationHelper.kt:
```diff
@@ -66,32 +68,42 @@ object NavigationHelper {
+            val (name, node) = locations.first()
+            node.pathFind(label = name, allowRerouting = true, condition = { true })
+            sendNavigateMessageWithContent("§7Only one location found, navigating to §r$name", goBack)
+        } else {
```
early return

rueblimaster replied:
doesn't really align with my coding style standards, but I don't want to make a discussion out of this so ye sure

hannibal002 reviewed (changes requested):
i like the improvement of the cmd, and the config merge!

rueblimaster committed:
pathfinger

rueblimaster committed:
Merge branch 'beta' into shnavigate-improvements

rueblimaster committed:
Reorder allowInstantNavigation in NavigationConfig

Moved the allowInstantNavigation property to the end of the NavigationConfig class and updated its description to include formatting for '/shnavigate'. No functional changes were made.

rueblimaster committed:
code design

rueblimaster committed:
Remove unused suffix logic from IslandAreas

Cleaned up the IslandAreas.kt file by removing commented-out code and the unused 'suffix' variable in the area display text. This simplifies the code and improves readability.

rueblimaster committed:
Update NavigationConfig.kt

hannibal002 committed:
Merge branch 'beta' into fork/rueblimaster/shnavigate-improvements

rueblimaster commented:
ups

hannibal002 committed:
different category structure, rename

rueblimaster committed:
do config fix

rueblimaster committed:
Update IslandAreas.kt

rueblimaster committed:
wrong config fix

hannibal002 committed:
config desc

rueblimaster committed:
fix config migration

rueblimaster committed:
Merge branch 'shnavigate-improvements' of https://github.com/rueblimaster/SkyHanni into shnavigate-improvements

hannibal002 committed:
fixed future feature. adds to config migration

hannibal002 committed:
split function in two
